flow/shared/postgres.go

```diff
@@ -150,3 +150,11 @@ func IsSQLStateError(err error, sqlStates ...string) bool {
 	var pgErr *pgconn.PgError
 	return errors.As(err, &pgErr) && slices.Contains(sqlStates, pgErr.Code)
 }
+
+func ConstructExcludedColumnsList(tableMappings []*protos.TableMapping) map[string][]string {
+	excludedColumns := make(map[string][]string)
+	for _, tm := range tableMappings {
+		excludedColumns[tm.SourceTableIdentifier] = slices.Clone(tm.Exclude)
```

Comment:
```suggestion
		excludedColumns[tm.SourceTableIdentifier] = tm.Exclude
```
`Exclude` should never be mutated (I mean should as in people should not do that, code today does not do that, because doing that would be bad)